### PR TITLE
feat(calibration): add CalibratedClassifier wrapper with online Platt scaling

### LINKED ISSUE #1421: Platt’s scaling for probability calibration
- [https://www.unofficialgoogledatascience.com/2021/04/why-model-calibration-matters-and-how.html](https://www.unofficialgoogledatascience.com/2021/04/why-model-calibration-matters-and-how.html)

### PR DESCRIPTION
**Pull request by Shizoqua**
Closes #1421.

Adds `calibration.CalibratedClassifier`, a wrapper that recalibrates a binary classifier's probability estimates using online Platt scaling.

## What it does

- Fits a logistic function `σ(a·s + b)` on top of the wrapped model's score `s`.
- The score is read **out-of-sample**: `predict_proba_one` is called on the wrapped model *before* it learns on the current sample, which prevents the calibration from overfitting to the same points it scores.
- `a` and `b` are updated online with one stochastic gradient step per sample, minimizing log-loss. They start at `a=1, b=0` (the identity), so the wrapper is a no-op until data flows in.
- Because the sigmoid is monotonic, the predicted label never changes — only the probabilities are spread/pulled.
- `predict_proba_one` always returns a distribution keyed on `False` and `True`.

## Implementation notes

- Mirrors the approach of scikit-learn's `_SigmoidCalibration` (Platt, 1999) but in the online setting, using `river.optim`-style gradient steps.
- Uses `utils.math.sigmoid` for numerical stability and a clamped `logit` for extreme probabilities.
- Passes the wrapper/classifier conventions: `base.Wrapper`, `base.Classifier`, `_unit_test_params` provided.

## Tests

- `tests/calibration/test_calibrated_classifier.py`:
  - identity mapping initially (`a=1, b=0`),
  - `predict_proba_one` returns a valid distribution,
  - calibration never changes the predicted label,
  - log-loss improves over the raw model on `Phishing`,
  - parameters move away from the identity after learning.

## Notes

This PR was produced with the assistance of an AI coding agent (opencode), in line with the project's coding-agent rules. The commit carries the author's `Signed-off-by`.

### CODE REVIEW

**Review by MaxHalford — COMMENTED** [review]
Good stuff, but this one definitely needs some appropriate documentation

**Comment by Shizoqua** [comment]
Thanks @MaxHalford! I added a \Use case\ section to the docstring explaining when calibration helps (probabilities consumed downstream, thresholds, expected value) and clarifying that calibration only rescales probabilities and never flips the predicted label. New commit 391caaad.

**Comment by Shizoqua** [comment]
Hi @MaxHalford, I've rebased onto the latest `upstream/main` and all checks are green. The use-case docstring we discussed is still in place. Could you re-review when you have time?

File: river/calibration/calibrated_classifier.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        The binary classifier to wrap.
+    lr
+        Learning rate used to update $a$ and $b$.
+
+    Attributes
+    ----------
+    a
+        Slope of the fitted sigmoid.
+    b
+        Intercept of the fitted sigmoid.
```
Comment (MaxHalford) [inline river/calibration/calibrated_classifier.py:63]:
```suggestion
    slope
        Slope of the fitted sigmoid.
    intercept
        Intercept of the fitted sigmoid.
```

I'll let you rename the code too

File: river/calibration/calibrated_classifier.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    >>> from river import preprocessing
+
+    >>> dataset = datasets.Phishing()
+
+    >>> model = preprocessing.StandardScaler() | linear_model.LogisticRegression()
+
+    >>> metric = metrics.LogLoss()
+
+    >>> evaluate.progressive_val_score(dataset, model, metric)
+    LogLoss: 0.3301120464388312
```
Comment (MaxHalford) [inline river/calibration/calibrated_classifier.py:82]:
```suggestion
    LogLoss: 0.330112...
```

File: river/calibration/calibrated_classifier.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+    The calibrated version spreads the probabilities further apart when the model is confident and
+    pulls them back towards 0.5 when it is not, which typically yields a lower log-loss:
+
+    >>> wrapped = preprocessing.StandardScaler() | linear_model.LogisticRegression()
+    >>> model = calibration.CalibratedClassifier(wrapped)
+
+    >>> evaluate.progressive_val_score(dataset, model, metric)
+    LogLoss: 0.3048718553325107
```
Comment (MaxHalford) [inline river/calibration/calibrated_classifier.py:91]:
```suggestion
    LogLoss: 0.304871...
```

File: river/api.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
     anomaly,
     bandit,
     base,
+    calibration,
```
Comment (MaxHalford) [inline river/api.py:10]: For now I'd prefer this to live in the `utils` module

**Review by MaxHalford — COMMENTED** [review]
Nearly there!

**Comment by Shizoqua** [comment]
Thanks @MaxHalford, all four points are addressed in a3acbd42:

- `CalibratedClassifier` now lives in `river.utils` (the `calibration` module and its `api.py` entry are gone; benchmark, tests, docs nav and release notes moved accordingly).
- The fitted parameters are renamed `slope` / `intercept` in the code, not just the docstring.
- The doctest log-losses use your truncated values (`0.330112...` and `0.304871...`).

I also rebased onto the latest `main`. Ready for another look.
